module/module.go

```diff
@@ -368,6 +391,17 @@ func (m *Module) RemoveResource(ctx context.Context, req *pb.RemoveResourceReque
 	if err := utils.TryClose(ctx, comp); err != nil {
 		m.logger.Error(err)
 	}
+
+	if name.Subtype.ResourceType == resource.ResourceTypeComponent && name.Subtype != generic.Subtype {
+		genSvc, ok := m.services[generic.Subtype]
+		if !ok {
+			return nil, errors.New("no generic service")
```

Comment:
this probably is not an error right? a component may not implement generic

Reply:
This is more about the generic subtype service missing, which should ALWAYS be there if a component model was ever registered (since it's unknown what specific models may or may not support generic, the service is just there as long as any component has been registered.) In theory this should be impossible to hit, but "abundance of caution" and all that.

If a component doesn't actually implement generic, then what will happen is the model itself will return "Unimplemented" if/when a generic (DoCommand) is actually tried on the resource or the service will error when trying to cast the resource. (This is effectively the same way things work in the main viam-server process.)

Reply:
Oh I see. Okay great then. Approving.